test/Microsoft.AspNet.Mvc.ViewFeatures.Test/Rendering/BufferedHtmlContentTest.cs

```diff
@@ -0,0 +1,127 @@
+        public void AppendString_AppendsAString()
+        {
+            // Arrange
+            var content = new BufferedHtmlContent();
+            var writer = new StringWriter();
+
+            // Act
+            content.Append("Hello");
+
+            // Assert
+            var result = Assert.Single(content.Entries);
+            Assert.IsType(typeof(string), result);
+        }
+
+        [Fact]
+        public void AppendCharArray_AppendsAsString()
+        {
+            // Arrange
+            var content = new BufferedHtmlContent();
+            var writer = new StringWriter();
+
+            // Act
+            content.Append(new char[] { 'h', 'e', 'l', 'l', 'o' }, 0, 5);
+
+            // Assert
+            var result = Assert.Single(content.Entries);
+            Assert.IsType(typeof(string), result);
+        }
+
+        [Fact]
+        public void AppendIHtmlContent_AppendsAsIs()
+        {
+            // Arrange
+            var content = new BufferedHtmlContent();
+            var writer = new StringWriter();
+
+            // Act
+            content.Append(new TestHtmlContent("Hello"));
+
+            // Assert
+            var result = Assert.Single(content.Entries);
+            var testHtmlContent = Assert.IsType<TestHtmlContent>(result);
+            testHtmlContent.WriteTo(writer, new CommonTestEncoder());
+            Assert.Equal("Written from TestHtmlContent: Hello", writer.ToString());
+            writer.Dispose();
+        }
+
+        [Fact]
+        public void CanAppendMultipleItems()
+        {
+            // Arrange
+            var content = new BufferedHtmlContent();
+            var writer = new StringWriter();
+
+            // Act
+            content.Append(new TestHtmlContent("hello"));
+            content.Append("Test");
+
+            // Assert
+            Assert.Equal(2, content.Entries.Count());
+            content.WriteTo(writer, new CommonTestEncoder());
+            Assert.Equal("Written from TestHtmlContent: helloTest", writer.ToString());
+        }
+
+        [Fact]
+        public void Clear_DeletesAllItems()
+        {
+            // Arrange
+            var content = new BufferedHtmlContent();
+            content.Append(new TestHtmlContent("hello"));
+            content.Append("Test");
+
+            // Act
+            content.Clear();
+
+            // Assert
+            Assert.Equal(0, content.Entries.Count());
+        }
+
+        [Fact]
+        public void WriteTo_WritesAllItems()
+        {
+            // Arrange
+            var content = new BufferedHtmlContent();
+            var writer = new StringWriter();
+            content.Append(new TestHtmlContent("Hello"));
+            content.Append("Test");
+
+            // Act
+            content.WriteTo(writer, new CommonTestEncoder());
+
+            // Assert
+            Assert.Equal(2, content.Entries.Count());
+            Assert.Equal("Written from TestHtmlContent: HelloTest", writer.ToString());
```

Comment:
I'll make the other test not use a writer.. They are similar but with different purposes.